### PR TITLE
add ideas regarding proof of capacity

### PR DESCRIPTION
**Pull request by zaibon**

### CODE REVIEW

File: specs/proof_of_capacity.md [hunk 1]
```diff
@@ -21,4 +21,27 @@ This shows that today, the validity of the capacity registered is only based on
...
+- CPU
+- Memory
+- Hard drive disk storage
+- Solid state drive storage
+
+### CPU and Memory
+For CPU and memory, we could use some kind of computational puzzle that are CPU or memory bound to ensure that the capacity a node reports can indeed be verified by solving one or more of these puzzles.  
+For this to work the difficulty of a puzzle need to be tunable and link with a fix minimum amount of CPU power or amount of memory.
+
+On problem with this approach is that once the node will be used to run workloads, both the CPU and memory will be used and not available to solve puzzle anymore. So we don't have a solution that allow us to re-check the capacity of a node over time while it is being used.
```
Comment (GlenDC) [inline specs/proof_of_capacity.md]: I don't think this is an issue, as we should only pay for what is used, once it is used. And whatever is using should serve as proof of its own. The thing running on it, or storing the stuff, should be able to proof it for itself in a different but still easy way. The storage use-case could easily request a certain range of bytes randomly at points, as to give an example.

File: specs/proof_of_capacity.md [hunk 2]
```diff
@@ -21,4 +21,27 @@ This shows that today, the validity of the capacity registered is only based on
...
+
+## Possible solutions
+
+We need to find a solution to measure the capacity of 4 units:
+
+- CPU
+- Memory
+- Hard drive disk storage
+- Solid state drive storage
+
```
Comment (GlenDC) [inline specs/proof_of_capacity.md]: You miss a fifth unit, network bandwith

File: specs/proof_of_capacity.md [hunk 3]
```diff
@@ -21,4 +21,27 @@ This shows that today, the validity of the capacity registered is only based on
...
+On problem with this approach is that once the node will be used to run workloads, both the CPU and memory will be used and not available to solve puzzle anymore. So we don't have a solution that allow us to re-check the capacity of a node over time while it is being used.
+
+### Disk storage
+For disk storage it also exists some kind of proof of capacity algorithm. Something like [burst](https://www.burst-coin.org/proof-of-capacity) is using:
+
+> Burst mining style solves all these problems (and is also ASIC-resistant) by allowing HDD mining – miners secure the network with their disk space. It can be seen as a “condensed Proof-of-Work”: you compute once (a process called plotting) and cache the results of your work on hard disk space. Then mining only requires to read through your cache – your HDD is idle most of the time and reads through the plot files only for a few seconds for each block.
+
+This also doesn't allow to do a capacity check while the node is being used cause it required to store the data on disk for a long period, which we can't do in our case.
```
Comment (GlenDC) [inline specs/proof_of_capacity.md]: As said in a previous comment, the storage use-case could easily request a certain range of bytes randomly at points, as to give an example.

File: specs/proof_of_capacity.md (hunk 1 above)
Comment (zaibon) [reply specs/proof_of_capacity.md]: >as we should only pay for what is used, once it is used

That's actually not how things works. There are 2 ways of getting token.
1. when you as a farmer provide capacity to the grid
2. when your capacity is used

This document is focusing on the first part, where we need some proof of the capacity that the farmer gives to the grid.

Also I'm not sure your example for storage stands, cause once the capacity is used, it is reserve and private to someone, so in the case of storage, the network has no way to know kind of data is stored and thus no way to validate that the farmer is still able to provide it.

**Comment by GlenDC** [comment]
Going to close this for now. Will leave the branch untouched. We can pick this up where we left if we ever do decide to take this anywhere.

**Pull request closed without merging**